python/ray/_private/function_manager.py

```diff
@@ -118,29 +111,52 @@ def compute_collision_identifier(self, function_or_class):
             dis.dis(function_or_class, file=string_file, depth=2)
         else:
             dis.dis(function_or_class, file=string_file)
-        collision_identifier = (
-            function_or_class.__name__ + ":" + string_file.getvalue())
+        collision_identifier = (function_or_class.__name__ + ":" +
+                                string_file.getvalue())
 
         # Return a hash of the identifier in case it is too large.
         return hashlib.sha1(collision_identifier.encode("utf-8")).digest()
 
+    def load_function_or_class_from_local(self, module_name,
+                                          function_or_class_name):
+        """Try to load a function or class in the module from local."""
+        module = importlib.import_module(module_name)
```

Comment:
This line may raises exceptions.

If the target object is defined in `__main__`, then we get the module name from it's code object, please refer to https://github.com/ray-project/ray/blob/master/python/ray/includes/function_descriptor.pxi#L288. But, the `opentelemetry` will patch a code object from `tracing_helper`, then you will get a incorrect module name.

```python
import ray
ray.init()
@ray.remote
def foo(): return 'foo'

foo.remote()  # Generate the function descriptor
print(foo._function_descriptor)  # Get an incorrect function descriptor
# output: {type=PythonFunctionDescriptor, module_name=tracing_helper, class_name=, function_name=foo, function_hash=47d95eadcfdd01484b125defd1b5b1403994dcb69422e90bfac573df}
```

Reply:
I think we need a better way to extract the module name from target, then add a try to handle the import error.

Reply:
talked offline, this is an existing issue. we can handle it in the next PR.